lib/review.lib.js

```diff
@@ -0,0 +1,84 @@
+const Review = require('../models/review.model');
+const ErrorResponse = require('../utils/errorResponse.util');
+const BookLib = require('./book.lib');
+const UserLib = require('./user.lib');
+
+class ReviewLib {
+  constructor() {
+    this.ReviewModel = Review;
+    this.bookLib = new BookLib();
+    this.userLib = new UserLib();
+  }
+
+  checkResourceInDB = async ({ user, book }) => {
+    if (user) {
+      const isUserInDB = await this.userLib.isUserInDB(user);
+      if (!isUserInDB) throw new ErrorResponse(`User with id: ${user} is not in database`, 422);
+    }
+    if (book) {
+      const isBookInDB = await this.bookLib.isBookInDB(book);
+      if (!isBookInDB) throw new ErrorResponse(`Book with id: ${book} is not in database`, 422);
+    }
+    return null;
+  };
+
+  createReview = async (reviewDetails) => {
+    const { ReviewModel } = this;
+    const {
+      reviewText, stars, user, book, likes,
+    } = reviewDetails;
+    let review;
+    try {
+      review = new ReviewModel({
+        reviewText, stars, user, book, likes,
+      });
+      return await review.save();
+    } catch (error) {
```

Comment:
You can as well move the review declaration inside the try-catch since it is not used anywhere outside